packages/router/src/lib/helpers/shared.ts

```diff
@@ -1,8 +1,95 @@
-import { getNtpTimeSeconds as _getNtpTimeSeconds } from "@connext/nxtp-utils";
+import { getNtpTimeSeconds as _getNtpTimeSeconds, RequestContext, GAS_ESTIMATES } from "@connext/nxtp-utils";
+import { BigNumber, constants } from "ethers";
+import { getOracleContractAddress, getPriceOracleInterface } from "../../adapters/contract/contract";
+import { getContext } from "../../router";
+import { ETHEREUM_CHAIN_ID } from "./auction";
 
 /**
  * Helper to allow easy mocking
  */
 export const getNtpTimeSeconds = async () => {
   return await _getNtpTimeSeconds();
 };
+
+/**
+ * Helper to calculate router gas fee in token
+ *
+ * @param sendingChainId The source chain Id
+ * @param receivingAssetId The asset address on destination chain
+ * @param receivingChainId The destination chain Id
+ * @param outputDecimals Decimal number of receiving asset
+ * @param requestContext Request context instance
+ */
+export const calculateGasFeeInReceivingToken = async (
+  sendingAssetId: string,
+  sendingChainId: number,
+  receivingAssetId: string,
+  receivingChainId: number,
+  outputDecimals: number,
+  requestContext: RequestContext,
+): Promise<BigNumber> => {
+  const chaindIdForGasFee = getChainIdForGasFee();
+  if (chaindIdForGasFee != sendingChainId && chaindIdForGasFee != receivingChainId) return constants.Zero;
```

Comment:
Need to handle the case where there are multiple chainIds for gas fees. Also we should add the gas fee estimation to rinkeby and goerli if we can so we can test this on testnet.